ocis/pkg/command/migrate.go

```diff
@@ -45,6 +58,146 @@ func init() {
+				err = errors.Wrap(err, "error creating a new manager")
+				return err
+			}
+			s, err := metadata.NewCS3Storage(conf.GatewayAddr, conf.ProviderAddr, conf.ServiceUserID, conf.ServiceUserIdp, conf.MachineAuthAPIKey)
+			if err != nil {
+				return err
+			}
+			err = s.Init(ctx, "jsoncs3-share-manager-metadata")
+			if err != nil {
+				return err
+			}
+			gc, err := pool.GetGatewayServiceClient(conf.GatewayAddr)
+			if err != nil {
+				return err
+			}
+			mgr, err := jsoncs3.New(s, gc, 0, nil, 1)
+			if err != nil {
+				return err
+			}
+
+			// Rebuild indexes
+			errorsOccured := false
+			storages, err := s.ReadDir(ctx, "storages")
+			if err != nil {
+				return err
+			}
+			for iStorage, storage := range storages {
+				fmt.Printf("Scanning storage %s (%d/%d)\n", storage, iStorage+1, len(storages))
+				spaces, err := s.ReadDir(ctx, filepath.Join("storages", storage))
+				if err != nil {
+					fmt.Printf("failed! (%s)\n", err.Error())
+					errorsOccured = true
+					continue
+				}
+
+				for iSpace, space := range spaces {
+					fmt.Printf("  Rebuilding space %d/%d...", iSpace+1, len(spaces))
+
+					spaceBlob, err := s.SimpleDownload(ctx, filepath.Join("storages", storage, space))
+					if err != nil {
+						fmt.Printf(" failed! (%s)\n", err.Error())
+						errorsOccured = true
+						continue
+					}
+					shares := &providercache.Shares{}
+					err = json.Unmarshal(spaceBlob, shares)
+					if err != nil {
+						fmt.Printf(" failed! (%s)\n", err.Error())
+						errorsOccured = true
+						continue
+					}
+					for _, share := range shares.Shares {
+						err = mgr.Cache.Add(ctx, share.ResourceId.StorageId, share.ResourceId.SpaceId, share.Id.OpaqueId, share)
+						if err != nil {
+							fmt.Printf(" failed! (%s)\n", err.Error())
+							errorsOccured = true
+						}
+						err = mgr.CreatedCache.Add(ctx, share.Creator.OpaqueId, share.Id.OpaqueId)
+						if err != nil {
+							fmt.Printf(" failed! (%s)\n", err.Error())
+							errorsOccured = true
+						}
+
+						spaceId := share.ResourceId.StorageId + shareid.IDDelimiter + share.ResourceId.SpaceId
+						switch share.Grantee.Type {
+						case provider.GranteeType_GRANTEE_TYPE_USER:
+							// TODO: figure out whether we need to upload
+							userid := share.Grantee.GetUserId().GetOpaqueId()
+
+							rs := &collaboration.ReceivedShare{
+								Share: share,
+								State: collaboration.ShareState_SHARE_STATE_PENDING,
+							}
+							err := mgr.UserReceivedStates.Add(ctx, userid, spaceId, rs)
+
+							if err != nil {
+								fmt.Printf(" failed! (%s)\n", err.Error())
+								errorsOccured = true
+							}
+						case provider.GranteeType_GRANTEE_TYPE_GROUP:
+							groupid := share.Grantee.GetGroupId().GetOpaqueId()
+							err := mgr.GroupReceivedCache.Add(ctx, groupid, spaceId)
+
+							if err != nil {
+								fmt.Printf(" failed! (%s)\n", err.Error())
+								errorsOccured = true
+							}
```

Comment:
we could check the error after the switch case

Reply:
With the improved logging it makes sense to keep it the way it is to get more meaningful logs imo.